addons/l10n_eu_service/models/res_company.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from .eu_tax_map import EU_TAX_MAP
+
+
+class Company(models.Model):
+    _inherit = 'res.company'
+
+    def _get_repartition_line_copy_values(self, original_rep_lines):
+        return [(0, 0, {
+            'factor_percent': line.factor_percent,
+            'repartition_type': line.repartition_type,
+            'account_id': line.repartition_type == 'tax' and line.account_id.id or None,
+            'company_id': line.company_id.id,
+            'sequence': line.sequence,
+        }) for line in original_rep_lines]
+
+    def map_eu_taxes(self):
+        eu_countries = self.env.ref('base.europe').country_ids
+        tax_group_id = self.env.ref('l10n_eu_service.tax_group_oss').id
+        for company in self:
+            tax_ids = self.env['account.tax'].search([('type_tax_use', '=', 'sale'),
+                ('amount_type', '=', 'percent'),
+                ('company_id','=', company.id)])
+            for domestic_tax in tax_ids:
```

Comment:
For efficiency with the way fiscal positions are searched (see comment below):

```suggestion

for country in eu_countries - company.account_tax_fiscal_country_id:
    # Search for or create the fiscal position
    
    for domestic_tax in tax_ids:
         # Do some stuff and fill in a list of (0,0, ...) ORM commands, unless the fiscal position already contains a mapping for domestic_tax
    
    # Update the fiscal position, using the list of commands you created
                        
```

Reply:
loop revised